### PR TITLE
feat: implemented emitImportedFiles flag

### LINKED ISSUE #294: Latest versions don't import google proto files automatically
Trying to generate the messages with `google.protobuf` types used- there are links on them on code, but they are no `.ts` files generated for them anymore. Saw no breaking changes in latest 5 versions

### PR DESCRIPTION
**Pull request by doochik**
Fixes #294.
This commit puts changes from #283 behind the flag.

### CODE REVIEW

File: src/plugin.ts [hunk 1]
```diff
@@ -24,8 +24,9 @@ async function main() {
   const utils = makeUtils(options);
   const ctx: Context = { typeMap, options, utils };
 
+  const filesToGenerate = options.emitImportedFiles ? request.protoFile : protoFilesToGenerate(request);
   const files = await Promise.all(
-    protoFilesToGenerate(request).map(async (file) => {
+    filesToGenerate.map(async (file) => {
```
Comment (stephenh) [inline src/plugin.ts:29]: Nice!

I've been bitten by this before, but can you update `integration/codegen.ts` as well? Either with the same logic you have here, or maybe just have it always use `request.protoFile`? I don't think we have an integration test that explicitly tests that `protoFilesToGenerate` codepath, which is fine I think...

**Review by stephenh — APPROVED** [review]
Thank you!

File: src/plugin.ts (hunk 1 above)
Comment (doochik) [reply src/plugin.ts:29]: HI!

I've updated `codegen.ts`. I think usage of `request.protoFile` should be enough fo tests.

**Pull request merged**